dali/imgcodec/decoders/jpeg/jpeg_mem.cc

```diff
@@ -114,11 +109,14 @@ uint8* UncompressLow(const void* srcdata, FewerArgsForCompiler* argball) {
   cinfo.client_data = &jpeg_jmpbuf;
   jerr.error_exit = CatchError;
   if (setjmp(jpeg_jmpbuf)) {
-    delete[] tempdata;
     return nullptr;
   }
 
   jpeg_create_decompress(&cinfo);
+  auto destroy_cinfo = AtScopeExit([&cinfo]() {
+    jpeg_destroy_decompress(&cinfo);
```

Comment:
this is crucial. Some paths ended up without calling `jpeg_destroy_decompress` which was the source of the leak